eng/pipelines/common/evaluate-paths-job.yml

```diff
@@ -0,0 +1,53 @@
+### Job used to evaluate changed paths on a pull request to emit variables to condition jobs based on paths.
+
+parameters:
+  # Object containing subset include and exclude paths in an array form.
+  # Scenarios:
+  #  1. exclude paths are specified
+  #     Will include all paths except the ones in the exclude list.
+  #  2. include paths are specified
+  #     Will only include paths specified in the list.
+  #  3. exclude + include:
+  #     1st we evaluate changes for all paths except ones in excluded list. If we can't find
+  #     any applicable changes like that, then we evaluate changes for incldued paths
+  #     if any of these two finds changes, then a variable will be set to true.
+  #  In order to consume this variable you need to reference it via: $[ dependencies.evaluate_paths.outputs['SetPathVars_<subset>.containschange'] ]
+  #
+  #  Array form example
+  #  paths:
+  #  - subset: coreclr
+  #    include:
+  #    - src/libraries/System.Private.CoreLib/*
+  #    exclude:
+  #    - src/libraries/*
+  #
+  #  This example will include ALL path changes under src/libraries/ except the ones under src/libraries/*!System.Private.CoreLib/*
+  paths: []
+
+
+jobs:
+- ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
+  - job: evaluate_paths
+    displayName: Evaluate Paths
+    pool:
+      vmImage: 'macOS-10.14'
```

Comment:
Unrelated to this PR but should we use a different vm for that given that OSX cores were decreased and we randomly hit the disconnect issues on it? Is there another image that is fast and reliable enough?

Reply:
I would rather stick with this pool for 2 reasons:

1) hosted pools are very fast to provide and bring your own cloud pools since those depend on the helix autoscaler can take up to 5-10 mins to provide for Linux (where the checkout is fast). 
2) I'm not worried about the disconnect issues as those happen on long jobs, this job should take less than 1 minute. Same for the core reduction, we don't need a lot of cores to run this job, it is very simple.